src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1382,21 +1352,63 @@ void walkInnerSymbols(ObjectType type, String innerNamespace) {
           privateEnums.add(innerNamespace + '.' + propName);
         }
       }
+      boolean nonEmpty = false;
 
       for (String propName : getSortedPublicPropertyNames(type)) {
         String qualifiedName = innerNamespace + '.' + propName;
         if (provides.contains(qualifiedName)) continue;
         JSType pType = type.getPropertyType(propName);
-
         if (pType.isEnumType()) {
+          if (!nonEmpty) {
+            emitNamespaceBegin(innerNamespace);
```

Comment:
Couldn't you have some function to check `hasAnyNamespaceMembers(type)`, and inspect that? That duplicates the logic a bit, but not too terribly so?

In any case, I'd call the boolean `foundNamespaceMembers` or so and invert the logic. Then you have `if (!foundNamespaceMembers) { emitNamespace(); foundNamespaceMembers = true; }` which would make this a lot more readable.